### PR TITLE
Add Linux support to Connectivity Plus

### PR DESCRIPTION
**Pull request by jpnurmi**

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Run this on my machine and worked, although I saw the following exception:

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: Bad state: StreamSink is closed
#0      _StreamSinkImpl.add (dart:io/io_sink.dart:154:7)
#1      _Socket.add (dart:io-patch/socket_patch.dart:1848:38)
#2      DBusClient._sendMessage (package:dbus/src/dbus_client.dart:707:13)
<asynchronous suspension>
#3      DBusClient._sendMethodCall (package:dbus/src/dbus_client.dart:655:5)
<asynchronous suspension>
```

As well, the e2e test did not pass when running on Linux.

Run the following test (inside the example project)
```
flutter drive --target=./test_driver/test/connectivity_e2e.dart
```

And got:
```
flutter: (The following exception is now available via WidgetTester.takeException:)
flutter: ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
flutter: The following StateError was thrown running a test:
flutter: Bad state: StreamSink is closed
flutter: 
flutter: When the exception was thrown, this was the stack:
flutter: #0      _StreamSinkImpl.add (dart:io/io_sink.dart:154:7)
flutter: #1      _Socket.add (dart:io-patch/socket_patch.dart:1848:38)
flutter: #2      DBusClient._sendMessage (package:dbus/src/dbus_client.dart:707:13)
flutter: <asynchronous suspension>
flutter: <asynchronous suspension>
flutter: (elided one frame from package:stack_trace)
flutter: 
flutter: The test description was:
flutter:   test connectivity result
```

**Comment by jpnurmi** [comment]
Hopefully fixed in dbus.dart soon :smile: 
- https://github.com/canonical/dbus.dart/issues/111
- https://github.com/canonical/dbus.dart/pull/114

**Comment by jpnurmi** [comment]
There were lots of last minute cleanup & fixup changes done collectively all over the plugins right before the release. I'll rebase this PR and try to catch up with all those changes later tonight or during the weekend.

**Comment by jpnurmi** [comment]
I tried pinning the dbus dependency to version 0.0.0-dev.19 to avoid https://github.com/canonical/dbus.dart/issues/111 (throws an exception starting from 0.0.0-dev.20):

```
[connectivity_plus_linux]: Package validation found the following potential issue:
[connectivity_plus_linux]: * Your dependency on "dbus" should allow more than one version. For example:
[connectivity_plus_linux]:   
[connectivity_plus_linux]:   dependencies:
[connectivity_plus_linux]: 
[connectivity_plus_linux]:     dbus: ^0.0.0-dev.19
[connectivity_plus_linux]:   
[connectivity_plus_linux]:   Constraints that are too tight will make it difficult for people to use your package
[connectivity_plus_linux]:   along with other packages that also depend on "dbus".
```

Then I tried with `dbus: ">=0.0.0-dev.18 <=0.0.0-dev.19"`:

```
[connectivity_plus_linux]: Package validation found the following potential issue:
[connectivity_plus_linux]: * Packages dependent on a pre-release of another package should themselves be published as a pre-release version. If this package needs dbus version 0.0.0-dev.18, consider publishing the package as a pre-release instead.
[connectivity_plus_linux]:   See https://dart.dev/tools/pub/publishing#publishing-prereleases For more information on pre-releases.
[connectivity_plus_linux]: 
[connectivity_plus_linux]: Package has 1 warning.
```

**Comment by jpnurmi** [comment]
For the failing test, looks like we should do something like here: https://github.com/flutter/plugins/commit/1685a2

File: packages/connectivity_plus_linux/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: connectivity_plus_linux
+description: Linux implementation of the connectivity_plus plugin
+homepage: https://plus.fluttercommunity.dev/
+repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
+version: 0.1.0
+
+environment:
+  sdk: ">=2.7.0 <3.0.0"
+  flutter: ">=1.17.0 <2.0.0"
```
Comment (jpnurmi) [inline packages/connectivity_plus_linux/pubspec.yaml:9]: update both

**Comment by miquelbeltran** [comment]
hi! whats stopping us from merging this one?

File: packages/connectivity_plus_linux/pubspec.yaml (hunk 1 above)
Comment (jpnurmi) [reply packages/connectivity_plus_linux/pubspec.yaml:9]: the problem vanished after rebasing on top of origin/main

**Pull request merged**